server/py/services/api/initial_data.py

```diff
@@ -877,40 +877,84 @@ def _perform_version_8_data_migrations(
 def _perform_version_9_data_migrations(
     db: framework.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
 ):
-    _migrate_function_kind(db, db_session, chunk_size=500)
+    _migrate_data(
+        db,
+        db_session,
+        framework.db.sqldb.models.Function,
+        "kind",
+        _handle_function_kind,
+    )
+    _migrate_data(
+        db,
+        db_session,
+        framework.db.sqldb.models.ArtifactV2,
+        "producer_uri",
+        _handle_artifact_producer_uri,
+    )
+
 
+def _handle_function_kind(records: list, record):
+    function_dict = record.struct
+    record.kind = function_dict.pop("kind", None)
+    if record.kind is not None:
+        record.struct = function_dict
+        records.append(record)
 
-def _migrate_function_kind(
-    db: framework.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session, chunk_size: int
+
+def _handle_artifact_producer_uri(records: list, record):
+    record.producer_uri = (
+        record.full_object.get("spec", {}).get("producer", {}).get("uri", None)
+    )
+    if record.producer_uri is not None:
+        records.append(record)
+
+
+def _migrate_data(
+    db: framework.db.sqldb.db.SQLDB,
+    db_session: sqlalchemy.orm.Session,
+    model,
+    field_name: str,
+    handle_field_record_func,
+    chunk_size: int = 500,
 ):
     offset = 0
-    number_of_functions_to_migrate = (
-        db._query(db_session, framework.db.sqldb.models.Function)
-        .filter(framework.db.sqldb.models.Function.kind.is_(None))
+    # Get count of records to migrate
+    count = (
+        db._query(db_session, model)
+        .filter(getattr(model, field_name).is_(None))
```

Comment:
If you generalizing the `_migrate_data` function I would accept the filter function as an argument too, and not limit to only checking if a field is `None`